File: my-class-mate-be/src/main/java/com/bill/model/response/UserResponse.java
package com.bill.model.response;

import com.bill.constant.RoleEnum;
import com.bill.model.StudentProfileDto;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class UserResponse {
    Long userId;
    String username;
    String nameTh;
    String surnameTh;
    String nameEn;
    String surnameEn;
    String email;
    RoleEnum role;
    Boolean isDeleted;
    StudentProfileDto studentProfile;
}

File: my-class-mate-fe/src/api/data/user-response.ts
import { Role } from "@/util/role-enum";

export interface UserRequest {
    userId?: string;
    username: string;
    password?: string;
    nameTh: string;
    surnameTh: string;
    nameEn: string;
    surnameEn: string;
    email: string;
    role: Role;
    isDeleted?: boolean;
    studentNo?: string;
}

export interface UserResponse {
    userId: string;
    username: string;
    password?: string;
    nameTh: string;
    surnameTh: string;
    nameEn: string;
    surnameEn: string;
    email: string;
    role: Role;
    isDeleted?: boolean;
    studentProfile?: StudentProfile;
    isUploadedImage?: boolean;
    imageCount: number;
}


export interface StudentProfile {
    studentNo?: string;
    address?: string | null;
    phoneNumber?: string | null;
    remark?: string | null;
}

export interface ImportResponse {
    updatedRow: number;
    createdRow: number;
}